hibernate-core/src/main/java/org/hibernate/hql/internal/ast/util/JoinProcessor.java

```diff
@@ -154,9 +157,7 @@ public static JoinType toHibernateJoinType(int astJoinType) {
 				if ( entityPersister != null && entityPersister instanceof AbstractEntityPersister ) {
 					AbstractEntityPersister aep = (AbstractEntityPersister) entityPersister;
 					String[] tables = aep.getTableNames();
-					for ( String table : tables ) {
```

Comment:
That was on purpose. I wanted to avoid the creation of the intermediate array list.

Reply:
Switched to `Collections.addAll` which also avoids the creation of the intermediate array list. (Aligned with the transition introduced in #2995 )